src/pyscipopt/scip.pyx

```diff
@@ -569,6 +570,11 @@ cdef class Solution:
 
             vals[name] = SCIPgetSolVal(self.scip, self.sol, scip_var)
         return str(vals)
+    
+    def _checkStage(self):
+        if self.sol == NULL and not SCIPgetStage(self.scip) == SCIP_STAGE_SOLVING:
+            raise Warning(f"method cannot only be called in stage SOLVING without a valid solution")
```

Comment:
For more informative error messages, we could pass the current method name to `_checkStage`,
`method X cannot be called` would be more helpful

Reply:
agreed. Please see my updated code.